src/marqo/client.py

```diff
@@ -204,48 +218,81 @@ def health(self):
 
     def _marqo_minimum_supported_version_check(self):
         min_ver = minimum_supported_marqo_version()
+        url = self.config.instance_mapping.get_control_url()
         skip_warning_message = (
-            f"Marqo encountered a problem trying to check the Marqo version found at `{self.url}`. "
+            f"Marqo encountered a problem trying to check the Marqo version found at `{url}`. "
             f"The minimum supported Marqo version for this client is {min_ver}. "
             f"If you are sure your Marqo version is compatible with this client, you can ignore this message. ")
 
         # Skip the check if the url is previously labelled as "_skipped"
-        if self.url in marqo_url_and_version_cache and marqo_url_and_version_cache[self.url] == "_skipped":
+        if url in marqo_url_and_version_cache and marqo_url_and_version_cache[url] == "_skipped":
             mq_logger.warning(skip_warning_message)
             return
 
         # Skip the check for Marqo CloudV2 APIs right now
         skip_version_check_url = ["https://api.marqo.ai", "https://cloud.marqo.ai"]
-        if self.url in skip_version_check_url:
-            marqo_url_and_version_cache[self.url] = "_skipped"
+        if any(url.startswith(skip_url) for skip_url in skip_version_check_url):
+            marqo_url_and_version_cache[url] = "_skipped"
             mq_logger.warning(skip_warning_message)
             return
 
         # Do version check
         try:
-            if self.url not in marqo_url_and_version_cache:
-                marqo_url_and_version_cache[self.url] = self.get_marqo()["version"]
-            marqo_version = marqo_url_and_version_cache[self.url]
+            if url not in marqo_url_and_version_cache:
+                marqo_url_and_version_cache[url] = self.get_marqo()["version"]
+            marqo_version = marqo_url_and_version_cache[url]
             if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(min_ver):
                 mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
                                   f"{minimum_supported_marqo_version()} to function properly, but your Marqo version is {marqo_version}. "
                                   f"Please upgrade your Marqo instance to avoid potential errors. "
                                   f"If you have already changed your Marqo instance but still get this warning, please restart your Marqo client Python interpreter.")
         except (MarqoWebError, RequestException, TypeError, KeyError) as e:
             mq_logger.warning(skip_warning_message)
-            marqo_url_and_version_cache[self.url] = "_skipped"
+            marqo_url_and_version_cache[url] = "_skipped"
         return
+
+    @staticmethod
+    def raise_error_for_cloud(function_name: str = None):
+        raise errors.BadRequestError(
+            f"The `client.{function_name}()` API is not supported on Marqo Cloud and will be removed in "
```

Comment:
I would rephrase the error as 
`mq.{function_name}()` is not supported on Marqo Cloud. Please use `mq.index(...).{function_name}()` instead. Check https://docs.marqo.ai/1.1.0/API-Reference/indexes/ for more details.

Note no need to repeat it'll be removed in 2.0.0 as the deprecation warning covers that.